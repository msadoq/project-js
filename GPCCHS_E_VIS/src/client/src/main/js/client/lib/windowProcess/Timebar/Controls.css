.controls{
  margin: 1px 0px 0px 1px;
}
.controlsUl{
  display: table;
  margin: 0px;
  padding: 0px;
}
.controlsUlRight{
  position: absolute;
  top: 2px;
  right: 4px;
}
.controlsLi{
  display: table-cell;
}
.controlButtonPlay:hover{
  background: #549edd;
  box-shadow: 0px 0px 6px #337ab7 inset;
  border: 1px solid #337ab7;
}
.glyphIcon{
  top: 0px;
}
.controlButtonActive, .controlButtonActive:hover{
  background: #60a7e5;
  box-shadow: 0px 0px 6px #337ab7 inset;
  border: 1px solid #337ab7;
}
.controlButtonDisabled{
  opacity: 0.65;
  cursor: default;
  line-height: 1.6em;
}
.speedPopover{
  width: 200px;
}
.speedInputEl{
  display: inline-block;
  margin-right: 10px;
  width: 100px;
}
