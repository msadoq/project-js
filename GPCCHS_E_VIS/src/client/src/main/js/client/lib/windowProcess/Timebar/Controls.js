import React, { PureComponent, PropTypes } from 'react';
import ControlsLeft from './ControlsLeft';
import ControlsRight from './ControlsRight';
import styles from './Controls.css';

export default class Controls extends PureComponent {

  static propTypes = {
    isPlaying: PropTypes.bool.isRequired,
    timebarRealTime: PropTypes.bool.isRequired,
    play: PropTypes.func.isRequired,
    pause: PropTypes.func.isRequired,
    updateSpeed: PropTypes.func.isRequired,
    toggleTimesetter: PropTypes.func.isRequired,
    restoreWidth: PropTypes.func.isRequired,
    goNow: PropTypes.func.isRequired,
    jump: PropTypes.func.isRequired,
    messages: PropTypes.arrayOf(
      PropTypes.shape({
        message: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
      })
    ),
    timebarUuid: PropTypes.string.isRequired,
    timebarMode: PropTypes.string.isRequired,
    timebarSpeed: PropTypes.number.isRequired,
    currentSessionExists: PropTypes.bool.isRequired,
    masterTimelineExists: PropTypes.bool.isRequired,
    masterTimeline: PropTypes.shape({
      color: PropTypes.string,
      id: PropTypes.string.isRequired,
      kind: PropTypes.string.isRequired,
      timelineId: PropTypes.string.isRequired,
      offset: PropTypes.number.isRequired,
      sessionId: PropTypes.number.isRequired,
    }),
    masterSessionId: PropTypes.number,
    switchToNormalMode: PropTypes.func.isRequired,
    switchToRealtimeMode: PropTypes.func.isRequired,
    switchToExtensibleMode: PropTypes.func.isRequired,
    switchToFixedMode: PropTypes.func.isRequired,
  }

  static defaultProps = {
    masterTimeline: null,
    masterSessionId: null,
    messages: [],
  }

  render() {
    return (
      <div
        className={styles.controls}
      >
        <ControlsLeft
          isPlaying={this.props.isPlaying}
          play={this.props.play}
          pause={this.props.pause}
          updateSpeed={this.props.updateSpeed}
          toggleTimesetter={this.props.toggleTimesetter}
          restoreWidth={this.props.restoreWidth}
          goNow={this.props.goNow}
          jump={this.props.jump}
          messages={this.props.messages}
          timebarUuid={this.props.timebarUuid}
          timebarSpeed={this.props.timebarSpeed}
          currentSessionExists={this.props.currentSessionExists}
          masterTimeline={this.props.masterTimeline}
          masterSessionId={this.props.masterSessionId}
        />
        <ControlsRight
          play={this.props.play}
          switchToNormalMode={this.props.switchToNormalMode}
          switchToRealtimeMode={this.props.switchToRealtimeMode}
          switchToExtensibleMode={this.props.switchToExtensibleMode}
          switchToFixedMode={this.props.switchToFixedMode}
          timebarMode={this.props.timebarMode}
          timebarUuid={this.props.timebarUuid}
          timebarRealTime={this.props.timebarRealTime}
          currentSessionExists={this.props.currentSessionExists}
          masterTimelineExists={this.props.masterTimelineExists}
          masterTimeline={this.props.masterTimeline}
          masterSessionId={this.props.masterSessionId}
        />
      </div>
    );
  }
}
