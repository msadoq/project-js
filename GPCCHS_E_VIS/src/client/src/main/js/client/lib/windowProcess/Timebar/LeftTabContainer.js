import { connect } from 'react-redux';
import { addAndMountTimeline, updateMasterId } from '../../store/actions/timebars';
import { unmountTimeline } from '../../store/actions/timebarTimelines';
import { updateTimebarId } from '../../store/actions/pages';
import {
  updateId,
  updateOffset,
  updateSessionId,
  updateColor,
} from '../../store/actions/timelines';
import LeftTab from './LeftTab';

export default connect(
  state => ({ sessions: state.sessions }),
  {
    updateMasterId,
    updateOffset,
    updateSessionId,
    updateColor,
    addAndMountTimeline,
    updateId,
    unmountTimeline,
    updateTimebarId,
  }

)(LeftTab);
