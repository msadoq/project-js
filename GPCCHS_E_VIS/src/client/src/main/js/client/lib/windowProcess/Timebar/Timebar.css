@import '../style/variables.css';

.timebarWrapperCollapsed{
  height: 36px;
  padding-top: 2px;
  text-align: center;
  background: rgba(0,0,0,0.4);
}
.viewportWrapper{
    padding-top: 24px;
    position: relative;
    height: calc(100% + 13px);
    overflow: hidden;
}
.viewportContainer{
    min-height: 55px;
    position: relative;
    background: #fff;
    border: 1px solid #bbb;
    background: #FAFAFA;
    height: calc(100% - 52px);
}
.visuWindow{
    top: 0px;
    position: absolute;
    background: linear-gradient(90deg,
      rgba(200, 126, 249,0.5) 0%,
      rgba(200, 126, 249,0.4) 4%,
      rgba(200, 126, 249,0.3) 8%,
      rgba(200, 126, 249,0.3) 92%,
      rgba(200, 126, 249,0.4) 96%,
      rgba(200, 126, 249,0.5) 100%);
    height: 100%;
    cursor: pointer;
    z-index: 1;
}
.viewportDragging{
    background: linear-gradient(90deg,
      rgba(200, 126, 209,0.7) 0%,
      rgba(200, 126, 209,0.6) 4%,
      rgba(200, 126, 209,0.5) 8%,
      rgba(200, 126, 209,0.5) 92%,
      rgba(200, 126, 209,0.6) 96%,
      rgba(200, 126, 209,0.7) 100%);
    cursor: move;
}
.viewportResizing{
    background: linear-gradient(90deg,
      rgba(190, 126, 230,0.7) 0%,
      rgba(190, 126, 230,0.6) 4%,
      rgba(190, 126, 230,0.5) 8%,
      rgba(190, 126, 230,0.5) 92%,
      rgba(190, 126, 230,0.6) 96%,
      rgba(190, 126, 230,0.7) 100%);
}
.formatedFullDate{
  position: absolute;
  left: 50%;
  transform: translate(-50%, 30%);
  color: #222;
}

/*
  Cursors
*/
.lower, .upper, .current, .slide{
    z-index: 4;
    cursor: col-resize;
    position: absolute;
    height: 100%;
    width: 3px;
    &.current{
      border-left: 2px solid var(--currentCursorColor);
    }
    &.lower{
        left: 0px;
        border-left: 1px solid var(--boundCursorsColor);
    }
    &.upper{
        right: 0px;
        border-right: 1px solid var(--boundCursorsColor);
    }
    &.slide{
        right: 0px;
        border-right: 1px solid var(--slideCursorColor);
    }
    &.lower:before, &.upper:before{
        width: 10px;
        height: 10px;
        content: '≡';
        color: #E45;
        transform: rotate(90deg);
        position: absolute;
        top: 48%;
    }
    &.lower:before{
        left: 2px;
    }
    &.upper:before{
        left: -8px;
    }
}



/*
  Handles
*/
.handle{
  width: 12px;
  height: 12px;
  border-radius: 6px;
  position: absolute;
  z-index: 12;
  cursor: pointer;
  filter: brightness(0.9);
  &:hover{
    filter: brightness(0.8);
  }
  &.undisplayed {
    display: none;
  }
  transition: width 0.1s;
  transition: height 0.1s;
  &.handleLower, &.handleUpper{
    top: 5px;
    background: var(--boundCursorsColor);
    &.handleLower{
      left: 0px;
      transform: translateX(-50%);
    }
    &.handleUpper{
      right: 0px;
      transform: translateX(50%);
    }
  }
  &.handleCurrent{
    top: 20px;
    background: var(--currentCursorColor);
    transform: translateX(-40%);
  }
  &.handleSlideLower, &.handleSlideUpper{
    top: 36px;
    background: var(--slideCursorColor);
    transform: translateX(-25%);
  }
}

/*
  Formatted Times
*/
.lowerFormattedTime, .upperFormattedTime, .currentFormattedTime, .slideLower, .slideUpper{
    font-size: 0.9em;
    font-weight: bold;
    top: -14px;
    position: absolute;
    user-select: none;
    transition: top 0.1s;
}
.moved{
  top: -24px;
}
.slideLower, .slideUpper{
    top: -28px;
    font-size: 0.8em;
    color: rgba(0,0,0,0.6);
    &.slideLower{
        left: 4px;
    }
    &.slideUpper{
        right: 4px;
    }
}

.lowerFormattedTime{
    transform: translateX(-100%);
    color: #d34;
}
.upperFormattedTime{
    right: 0;
    color: #d34;
    transform: translateX(100%);
}
.currentFormattedTime{
    color: #1A2;
    white-space: nowrap;
    z-index: 2;
    transform: translateX(-50%);
}
.arrowLeft, .arrowRight, .arrowExt{
    z-index: 11;
    position: absolute;
    color: #fff;
    border-radius: 2px;
    border: none;
    top: 5px;
    width: 20px;
    height: 15px;
    cursor: pointer;
    box-shadow: 0px 0px 5px #666;
    padding: 0;
    line-height: 4px;
    font-size: 1.1em;
}
.arrowLeft{
    left: 7px;
}
.arrowRight, .arrowExt{
    right: 7px;
}
.arrowExt{
  width: 35px;
  font-size: 1em;
  font-size: 0.8em;
  line-height: 0.70em;
  font-weight: 600;
}
.arrowLeftBring, .arrowRightBring{
  background: #444;
  top: 25px;
  padding: 0;
  line-height: 4px;
  font-size: 1.1em;
}
.arrowLeftBring:hover, .arrowRightBring:hover{
  background: #111;
}
.resizeTimebarContainer{
  margin: 0;
  height: 4px;
  cursor: row-resize;
}
.resizeTimebarContainer:before{
    width: 20px;
    height: 12px;
    line-height: 12px;
    text-align: center;
    content: '≡';
    color: #888;
    position: absolute;
    top: 0px;
    left: 50%;
    font-size: 1.2em;
    transform: translateX(-50%);
    z-index: 4;
}
.resizingTimebarContainer:before{
  color: #555;
}
.timelines{
  height: 100%;
  overflow-y: auto;
  overflow-x: hidden;
  box-shadow: 0px 0px 8px rgba(0,0,0,0.2) inset;
}

.timeline{
  width: 100%;
  margin-bottom: 2px;
  height: 18px;
  position: relative;
}
.timelineInfos{
  width: 90%;
  max-width: 200px;
  background-color: rgba(255,255,255,0.8);
  height: 100%;
  margin: 0px auto;
  padding-left: 6px;
  line-height: 20px;
}
.timelineOffset{
  position: absolute;
  height: 2px;
  bottom: 0px;
  background-color: #F45;
}
.rightTab{
  border-width: 1px;
  border-style: solid;
  width: 100%;
  height: calc(100% - 13px);
}
.play{ border-color: var(--borderPlay); }
.pause{ border-color: var(--borderPause); }
.realtime{ border-color: var(--borderRealtime); }
.acc{ border-color: var(--borderAcc); }
.desc{ border-color: var(--borderDesc); }
