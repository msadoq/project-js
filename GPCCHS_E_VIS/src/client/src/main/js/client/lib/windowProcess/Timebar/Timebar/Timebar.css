@import '../../style/variables.css';

.viewportWrapper{
    padding-top: 24px;
    position: relative;
    height: calc(100% + 13px);
    overflow: hidden;
}
.viewportContainer{
    min-height: 55px;
    position: relative;
    background: #fff;
    border: 1px solid #bbb;
    background: #FAFAFA;
    height: calc(100% - 52px);
}

.formatedFullDate{
  position: absolute;
  left: 50%;
  transform: translate(-50%, 30%);
  color: #222;
}

/*
  Cursors
*/

.slide{
    z-index: 4;
    cursor: col-resize;
    position: absolute;
    height: 100%;
    width: 3px;
    right: 0px;
    border-right: 1px solid var(--slideCursorColor);
}
.lower:before, .upper:before{
    width: 10px;
    height: 10px;
    content: '≡';
    color: #E45;
    transform: rotate(90deg);
    position: absolute;
    top: 48%;
}
.lower:before{
    left: 2px;
}
.upper:before{
    left: -8px;
}

/*
  Handles
*/
.handle{
  width: 12px;
  height: 12px;
  border-radius: 6px;
  position: absolute;
  z-index: 12;
  cursor: pointer;
  filter: brightness(0.9);
  &:hover{
    filter: brightness(0.8);
  }
  &.undisplayed {
    display: none;
  }
  transition: width 0.1s;
  transition: height 0.1s;
  &.handleLower, &.handleUpper{
    top: 5px;
    background: var(--boundCursorsColor);
  }
  &.handleLower{
    left: 0px;
    transform: translateX(-50%);
  }
  &.handleUpper{
    right: 0px;
    transform: translateX(50%);
  }
  &.handleCurrent{
    top: 20px;
    background: var(--currentCursorColor);
    transform: translateX(-40%);
  }
  &.handleSlideLower, &.handleSlideUpper{
    top: 36px;
    background: var(--slideCursorColor);
    transform: translateX(-25%);
  }
}

/*
  Formatted Times
*/
.lowerFormattedTime, .upperFormattedTime, .currentFormattedTime, .slideLower, .slideUpper{
    font-size: 0.9em;
    font-weight: bold;
    top: -14px;
    position: absolute;
    user-select: none;
    transition: top 0.1s;
}
.moved{
  top: -24px;
}
.slideLower, .slideUpper{
    top: -28px;
    font-size: 0.8em;
    color: rgba(0,0,0,0.6);
    &.slideLower{
        left: 4px;
    }
    &.slideUpper{
        right: 4px;
    }
}

.lowerFormattedTime{
    transform: translateX(-100%);
    color: #d34;
}
.upperFormattedTime{
    right: 0;
    color: #d34;
    transform: translateX(100%);
}
.currentFormattedTime{
    color: #1A2;
    white-space: nowrap;
    z-index: 2;
    transform: translateX(-50%);
}
.arrowLeft, .arrowRight, .arrowExt{
    z-index: 11;
    position: absolute;
    color: #fff;
    border-radius: 2px;
    border: none;
    top: 5px;
    width: 20px;
    height: 15px;
    cursor: pointer;
    box-shadow: 0px 0px 5px #666;
    padding: 0;
    line-height: 4px;
    font-size: 1.1em;
}
.arrowLeft{
    left: 7px;
}
.arrowRight, .arrowExt{
    right: 7px;
}
.arrowExt{
  width: 35px;
  font-size: 1em;
  font-size: 0.8em;
  line-height: 0.70em;
  font-weight: 600;
}
.arrowLeftBring, .arrowRightBring{
  background: #444;
  top: 25px;
  padding: 0;
  line-height: 4px;
  font-size: 1.1em;
}
.arrowLeftBring:hover, .arrowRightBring:hover{
  background: #111;
}
.resizeTimebarContainer{
  margin: 0;
  height: 4px;
  cursor: row-resize;
}
.resizeTimebarContainer:before{
    width: 20px;
    height: 12px;
    line-height: 12px;
    text-align: center;
    content: '≡';
    color: #888;
    position: absolute;
    top: 0px;
    left: 50%;
    font-size: 1.2em;
    transform: translateX(-50%);
    z-index: 4;
}
.resizingTimebarContainer:before{
  color: #555;
}
.timelines{
  height: 100%;
  overflow-y: auto;
  overflow-x: hidden;
  box-shadow: 0px 0px 8px rgba(0,0,0,0.2) inset;
}

.timeline{
  width: 100%;
  margin-bottom: 2px;
  height: 18px;
  position: relative;
}
.timelineInfos{
  width: 90%;
  max-width: 200px;
  background-color: rgba(255,255,255,0.8);
  height: 100%;
  margin: 0px auto;
  padding-left: 6px;
  line-height: 20px;
}
.timelineOffset{
  position: absolute;
  height: 2px;
  bottom: 0px;
  background-color: #F45;
}
.rightTab{
  border-width: 1px;
  border-style: solid;
  width: 100%;
  height: 100%;
}
.play{ border-color: var(--borderPlay); }
.pause{ border-color: var(--borderPause); }
.realtime{ border-color: var(--borderRealtime); }
.acc{ border-color: var(--borderAcc); }
.desc{ border-color: var(--borderDesc); }
