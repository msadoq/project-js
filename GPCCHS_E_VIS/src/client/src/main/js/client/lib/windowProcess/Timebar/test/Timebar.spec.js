import React from 'react';
import renderer from 'react-test-renderer';
import TimelinesFixture from '../../../../test/__mocks__/Timelines';
import TimebarFixture from '../../../../test/__mocks__/Timebar';
import Timebar from '../Timebar/Timebar';

const propsStub = {
  retrieveFormattedFullDateEl: () => null,
  toggleTimesetter: () => null,
  onVerticalScroll: () => null,
  updateCursors: () => null,
  updateViewport: () => null,
  pause: () => null,
  play: () => null,
  jump: () => null,
  setRealTime: () => null,
  openModal: () => null,
  isPlaying: true,
  visuWindow: TimebarFixture.visuWindow,
  slideWindow: TimebarFixture.slideWindow,
  viewport: {
    lower: 1480417084354,
    upper: 1480417204665,
  },
  timelines: TimelinesFixture,
  timebarSpeed: TimebarFixture.speed,
  timebarRealTime: TimebarFixture.realTime,
  timebarMode: TimebarFixture.mode,
  timebarUuid: TimebarFixture.timebarUuid,
  verticalScroll: 0,
  widthPx: 900,
};

test('Timebar renders correctly', () => {
  const tree = renderer.create(
    <Timebar {...propsStub} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
