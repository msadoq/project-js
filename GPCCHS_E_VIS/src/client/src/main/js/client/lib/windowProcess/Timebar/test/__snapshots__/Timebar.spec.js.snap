// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timebar renders correctly 1`] = `
<div
  className="ViewportWrapper viewportWrapper"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
>
  <div
    className="viewportContainer"
    onWheel={[Function]}
  >
    <div
      className="visuWindow"
      onMouseDown={[Function]}
      style={
        Object {
          "left": "0.8311791939224177%",
          "width": "16.05090141383581%",
        }
      }
    >
      <span
        className="lower"
        cursor="lower"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
      />
      <span
        className="current"
        cursor="current"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "left": "44.90186940085962%",
          }
        }
      />
      <span
        className="upper"
        cursor="upper"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
      />
      <span
        className="handle handleLower undisplayed"
        cursor="lower"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
      />
      <span
        className="handle handleCurrent undisplayed"
        cursor="current"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "left": "44.90186940085962%",
          }
        }
      />
      <span
        className="handle handleUpper undisplayed"
        cursor="upper"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
      />
      <span
        className="LowerFormattedTime lowerFormattedTime"
      >
        10:58:05.354
      </span>
      <span
        className="CurrentFormattedTime currentFormattedTime"
        style={
          Object {
            "left": "44.90186940085962%",
          }
        }
      >
        10:58:14.025
      </span>
      <span
        className="UpperFormattedTime upperFormattedTime"
      >
        10:58:24.665
      </span>
    </div>
    <div
      className="Timelines timelines"
      onScroll={[Function]}
    >
      <div
        className="timeline"
        style={
          Object {
            "background": "#fa1289",
          }
        }
      >
        <div
          className="timelineInfos"
        >
          Session 1
        </div>
        <span
          className="timelineOffset"
          style={
            Object {
              "left": "auto",
              "right": "50%",
              "width": "0.16623583878448356%",
            }
          }
        />
      </div>
      <div
        className="timeline"
        style={
          Object {
            "background": "#31b0d5",
          }
        }
      >
        <div
          className="timelineInfos"
        >
          Session 2
        </div>
        <span
          className="timelineOffset"
          style={
            Object {
              "left": "auto",
              "right": "50%",
              "width": "0.8311791939224177%",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="Scale timebarRuleContainer"
    onMouseDown={[Function]}
  >
    <div
      className="timebarRule"
      style={
        Object {
          "left": "-36.86612196723492%",
        }
      }
    >
      <div
        className="scaleBar"
        style={
          Object {
            "left": "0%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          10:57:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "16.623583878448354%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          10:57:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "33.24716775689671%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          10:58:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "49.87075163534506%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          10:58:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "66.49433551379342%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          10:58:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "83.11791939224177%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          10:59:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "99.74150327069012%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          10:59:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "116.36508714913847%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          10:59:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "132.98867102758683%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:00:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "149.6122549060352%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:00:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "166.23583878448355%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:00:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "182.8594226629319%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:01:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "199.48300654138023%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:01:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "216.1065904198286%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:01:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "232.73017429827695%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:02:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "249.3537581767253%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:02:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "265.97734205517366%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:02:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "282.600925933622%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:03:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "299.2245098120704%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:03:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "315.84809369051874%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:03:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "332.4716775689671%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:04:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "349.09526144741545%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:04:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "365.7188453258638%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:04:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "382.34242920431217%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:05:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "398.96601308276047%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:05:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "415.5895969612088%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:05:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "432.2131808396572%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:06:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "448.83676471810554%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:06:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "465.4603485965539%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:06:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "482.08393247500226%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:07:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "498.7075163534506%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:07:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "515.331100231899%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:07:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "531.9546841103473%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:08:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "548.5782679887957%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:08:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "565.201851867244%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:08:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "581.8254357456924%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:09:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "598.4490196241408%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:09:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "615.0726035025891%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:09:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "631.6961873810375%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:10:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "648.3197712594858%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:10:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "664.9433551379342%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:10:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "681.5669390163825%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:11:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "698.1905228948309%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:11:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "714.8141067732793%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:11:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "731.4376906517276%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:12:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "748.061274530176%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:12:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "764.6848584086243%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:12:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "781.3084422870727%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:13:00
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "797.9320261655209%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:13:20
        </span>
        <span
          className="scaleBar"
        />
      </div>
      <div
        className="scaleBar"
        style={
          Object {
            "left": "814.5556100439693%",
          }
        }
      >
        <span
          className="ScaleTime scaleTime"
        >
          11:13:40
        </span>
        <span
          className="scaleBar"
        />
      </div>
    </div>
  </div>
</div>
`;
